### DISCUSSION TITLE
How to use a variable value inside Jinja template

### DISCUSSION THREAD

**Discussion by saikarthikp9** [apache/airflow#20313] [Q&A] [answered]
I am using jinja template when pulling an xcom value as seen below. The task_id is dynamic, how can I set that in the below statement?

example tas_id: f'S3ObjectList-{i}', where i can be any number.
 
'BUCKET_NAME': "{{ task_instance.xcom_pull(task_ids='S3ObjectList', key='return_value')['BUCKET_NAME'] }}",

**Comment by rsotogar** [comment, marked as answer]
Use 4 curly brackets both before and after the jinja template when using Python f strings:

'BUCKET_NAME':  f"{{{{ task_instance.xcom_pull(task_ids='S3ObjectList-{i}', key='return_value')['BUCKET_NAME'] }}}}"
